app/src/main/java/org/mozilla/tv/firefox/utils/PeriodicRequester.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.utils
+
+import android.support.annotation.VisibleForTesting
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+private const val CACHE_UPDATE_FREQUENCY_MINUTES = 45L
+private val CACHE_UPDATE_FREQUENCY_SECONDS = TimeUnit.MINUTES.toSeconds(CACHE_UPDATE_FREQUENCY_MINUTES)
+
+/**
+ * Simple interface for making a network request.
+ *
+ * This may operate on any thread, and consumers should take care to schedule
+ * it off of the main thread.
+ */
+interface Endpoint<T> {
+    fun request(): Single<Response<T>>
+}
+
+sealed class Response<T> {
+    data class Success<T>(val data: T) : Response<T>()
+    class Failure<T> : Response<T>()
+}
+
+/**
+ * Used to make long term, reoccurring network requests.
+ *
+ * Will make an initial request followed by periodic updates, with exponentially
+ * backing off retry attempts upon failure.
+ */
+class PeriodicRequester<T>(private val endpoint: Endpoint<T>) {
+
+    /**
+     * Makes one network request upon subscription. If this fails, more requests
+     * will be made on an exponential backoff.
+     *
+     * Follow up requests will be made every 45 minutes, regardless of previous
+     * success or failure.
+     *
+     * Repeated calls to this method will create separate [Observable]s,
+     * potentially duplicating calls.
+     */
+    fun start(): Observable<Response<T>> = normalTimer()
+        .flatMap {
+            Observable.concat(requestAsynchronously().toObservable(), backoffRequests())
+                .takeUntil { it is Response.Success }
```

Comment:
So essentially we request forever until the 200 response (with backoff logic)? Maybe I am not understanding your backoff logic - is the backoff interval exponentially incremental? So we could get to a point `Integer.MAX_VALUE` ^ 2?

This should be okay, but generally this is a good place where we can manipulate tcp headers (i.e. etags) and/or http status codes. Maybe something we can work with Pocket API on

Reply:
Yeah, the backoff is exponentially incremental.  It will cap out at `cache_update_frequency / 2` though, so we'll never actually reach `Integer.MAX_VALUE` ^ 2`.

Maybe we could talk later about how we can accomplish this with networking flags?  I'm not super knowledgable on that front.